pkg/kn/config/config.go

```diff
@@ -145,10 +144,12 @@ func BootstrapConfig() error {
 			return fmt.Errorf("cannot stat configfile %s: %w", configFile, err)
 		}
 		if err := os.MkdirAll(filepath.Dir(viper.ConfigFileUsed()), 0775); err != nil {
-			return err
+			// Can't create config directory, proceed silently without reading the config
+			return nil
 		}
 		if err := os.WriteFile(viper.ConfigFileUsed(), []byte(configContentDefaults), 0600); err != nil {
-			fmt.Fprintf(os.Stderr, "WARNING: failed writing config file to %q: %s\n", configFile, err)
```

Comment:
@rhuss let me know if you want to go without any warning completely.

Reply:
sounds good to me